### PR TITLE
feat: implement detectDistributors main method

### LINKED ISSUE #109: Feat: Implement detectDistributors main method
## Context

The detectDistributors method is the main entry point for the Distributor Detector component. It orchestrates the incremental scanning process, manages state persistence, and coordinates all the static helper methods to discover new distributors.

## Current Behavior

1. No main detection method exists
2. Cannot perform incremental scanning from last processed block
3. No way to discover distributors up to a specific date

## Expected Behavior

Implement `detectDistributors` async method that:
- Takes an endDate parameter (Date object)
- Reads existing distributor data and last_scanned_block from FileManager
-  Gets corresponding block number from date
- Determines scan range (last_scanned_block + 1 to end_date_block)
- Calls scanBlockRange if there are new blocks to scan
- Updates last_scanned_block in metadata
- Writes updated data back through FileManager
- Returns the complete DistributorsData object

Follow the algorithm in specification lines 178-218.

## Required Context

- `docs/specs/distributor-detector.md` - algorithm details (lines 178-218)
- `src/file-manager.ts` or similar - FileManager API for reading/writing
- `__tests__/test-data/distributor-detector` - test data and fixtures
- Look for date formatting patterns in codebase

## Blocking Issues

- Blocked by #107 - Need scanBlockRange method
- Blocked by #108 - Need DistributorDetector class structure

## Out of Scope

- Chain ID verification (mentioned in spec but can be added later)
- Error recovery strategies beyond basic error propagation

### PR DESCRIPTION
**Pull request by yahgwai**
## Summary
- Implements the main `detectDistributors` async method that orchestrates incremental scanning
- Follows TDD principles with comprehensive test coverage before implementation
- Refactored implementation to improve readability with extracted helper methods

## Implementation Details
The `detectDistributors` method:
- Takes an endDate parameter and finds the corresponding block number  
- Reads existing distributor data and last_scanned_block from FileManager
- Performs incremental scanning from the last processed block
- Updates metadata with new last_scanned_block and chain_id
- Merges new distributors with existing ones (skipping duplicates)
- Writes updated data back through FileManager

## Test Plan
- [x] Unit tests for all scenarios (first run, incremental, edge cases)
- [x] Error handling tests (missing dates, RPC failures, write errors)
- [x] Chain ID handling tests
- [x] All existing tests still pass
- [x] TypeScript compilation passes
- [x] ESLint checks pass
- [x] Code is properly formatted

## Notes
- Chain ID verification (mentioned in spec) was deferred as noted in the issue
- Error recovery strategies beyond basic error propagation were not implemented as per scope

Resolves #109

### CODE REVIEW

File: src/distributor-detector.ts [hunk 1]
```diff
@@ -189,4 +190,119 @@ export class DistributorDetector {
...
+    lastScannedBlock: number,
+  ): Promise<DistributorsData> {
+    // Get chain ID from provider
+    const network = await this.provider.getNetwork();
+    const chainId = Number(network.chainId);
+
+    // Initialize with existing distributors or empty object
+    const updatedData: DistributorsData = {
+      metadata: {
+        chain_id: chainId,
```
Comment (yahgwai) [inline src/distributor-detector.ts:297]: Dont overwrite if we have a chain id already
